app/src/org/commcare/views/dialogs/AlertDialogFragment.java

```diff
@@ -51,6 +54,9 @@ public void onDismiss(DialogInterface dialog) {
     @Override
     @NonNull
     public Dialog onCreateDialog(Bundle savedInstanceState) {
+        if (underlyingDialog == null) {
+            return super.onCreateDialog(savedInstanceState);
```

Comment:
hat makes this the right thing to do here?

Reply:
This just creates an empty dialogue which gets dismissed in onCreate. I can't return null from here since DialogFragment assumes that this method always returns a Non null value, so to get past the NPE I couldn't think of any other solution.  Also I can't dismiss the dialogue from here since it's not been created yet.

Reply:
Hm, what's confusing me then is that [the docs](https://developer.android.com/reference/android/support/v4/app/DialogFragment.html#onCreateDialog(android.os.Bundle)) say that this method gets called AFTER `onCreate()`. I guess the issue is just that calling `dismiss()` in `onCreate()` doesn't stop `onCreateDialog()` from getting called?

Reply:
And yeah, you are right. `onCreateDialog()` gets called even if we dismiss the dialog in `onCreate()` and results in NPE. I am going to move even `dismiss()` inside onCreateDialog() just to make it bit less confusing.